app/code/Magento/Ui/view/base/web/js/lib/knockout/bindings/scope.js

```diff
@@ -13,6 +13,43 @@ define([
 ], function (ko, registry, $t, renderer, $, consoleLogger) {
     'use strict';
 
+    /**
+     * Check if an element is visible in the viewport
+     *
+     * @param {HTMLElement} el
+     * @returns bool
+     */
+    function isInViewport(el) {
+        if (!el.checkVisibility()) return false;
+        const rect = el.getBoundingClientRect();
+        const vWidth = window.innerWidth || doc.documentElement.clientWidth;
+        const vHeight = window.innerHeight || doc.documentElement.clientHeight;
+
+        // Check if the element is out of bounds
+        if (rect.right < 0 || rect.bottom < 0 || rect.left > vWidth || rect.top > vHeight) return false;
+
+        // Return true if any of the above disjunctions are false
+        return true;
+    }
+
+    /**
+     *  Check condition apply components
+     *
+     * @param {HTMLElement} el - element to apply bindings to.
+     * @param {ko.bindingContext} bindingContext - instance of ko.bindingContext, passed to binding initially.
+     * @param {Promise} promise - instance of jQuery promise
+     * @param {Object} component - component instance to attach to new context
+     */
+    function applyComponents(el, bindingContext, promise, component) {
+        if (isInViewport(el)) {
+            runApplyComponents(el, bindingContext, promise, component);
+        } else {
+            document.addEventListener('mousemove', function (e) {
```

Comment:
@rogerdz 
what about touch events? 
I would suggest replacement for this one, `lib/web/mage/apply/main.js::applyFor` and `lib/web/mage/apply/main.js::lazyProcess` for following form:

```js
(events => {
    const lazyLoadJs = () => {
        events.forEach(type => window.removeEventListener(type, lazyLoadJs))
        runApplyComponents(el, bindingContext, promise, component);
    };
    events.forEach(type => window.addEventListener(type, lazyLoadJs, {once: true}))
})(['touchstart', 'mouseover', 'wheel', 'scroll', 'keydown'])
```